services/src/main/java/com/broadleafcommerce/bulkoperations/service/handler/CatalogBulkOperationHandler.java

```diff
@@ -0,0 +1,171 @@
+ * LLC and may be covered by U.S. and Foreign Patents, patents in process, and are protected by
+ * trade secret or copyright law. Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained from Broadleaf Commerce, LLC.
+ */
+package com.broadleafcommerce.bulkoperations.service.handler;
+
+import static com.broadleafcommerce.bulkoperations.service.environment.RouteConstants.Persistence.BULK_OPS_ROUTE_KEY;
+import static com.broadleafcommerce.common.messaging.service.DefaultMessageLockService.MESSAGE_IDEMPOTENCY_KEY;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.Message;
+
+import com.broadleafcommerce.bulk.v2.domain.BulkOperationRequest;
+import com.broadleafcommerce.bulk.v2.domain.BulkOperationResponse;
+import com.broadleafcommerce.bulk.v2.domain.SupportedBulkOp;
+import com.broadleafcommerce.bulk.v2.messaging.BulkOpsInitializeItemsRequest;
+import com.broadleafcommerce.bulk.v2.messaging.BulkOpsInitializeItemsRequestProducer;
+import com.broadleafcommerce.bulk.v2.messaging.sandbox.CreateSandboxRequest;
+import com.broadleafcommerce.bulk.v2.messaging.sandbox.CreateSandboxRequestProducer;
+import com.broadleafcommerce.bulkoperations.domain.CatalogItem;
+import com.broadleafcommerce.bulkoperations.service.environment.BulkOperationsProviderProperties;
+import com.broadleafcommerce.bulkoperations.service.provider.CatalogProvider;
+import com.broadleafcommerce.common.extension.TypeFactory;
+import com.broadleafcommerce.common.messaging.notification.DetachedDurableMessageSender;
+import com.broadleafcommerce.data.tracking.core.context.ContextInfo;
+
+import java.security.SecureRandom;
+import java.util.List;
+import java.util.Random;
+
+import io.azam.ulidj.ULID;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+
+public class CatalogBulkOperationHandler implements BulkOperationHandler {
+
+    @Getter(AccessLevel.PROTECTED)
+    private final CatalogProvider<? extends CatalogItem> catalogProvider;
+
+    @Getter(value = AccessLevel.PROTECTED)
+    private final DetachedDurableMessageSender sender;
+
+    @Getter(AccessLevel.PROTECTED)
+    @Setter(onMethod_ = {@Autowired})
+    private BulkOperationsProviderProperties properties;
+
+    @Getter(AccessLevel.PROTECTED)
+    private final CreateSandboxRequestProducer createSandboxRequestProducer;
+
+    @Getter(AccessLevel.PROTECTED)
+    private final BulkOpsInitializeItemsRequestProducer bulkOpsInitializeItemsRequestProducer;
+
+    @Getter(AccessLevel.PROTECTED)
+    private static final Random SECURE_RANDOM = new SecureRandom();
```

Comment:
Should be fine to just make it protected and not need a getter given that it is final and static
```suggestion
    protected static final Random SECURE_RANDOM = new SecureRandom();
```